packages/provider/src/tasks/client/clientTasks.ts

```diff
@@ -67,81 +67,69 @@ export class ClientTaskManager {
-					commitment: UserCommitment | PoWCaptchaStored,
-				): boolean => {
-					return !!lastTask.updated && isCommitmentUpdated(commitment);
-				};
-
-				commitments = commitments.filter((commitment) =>
-					commitmentUpdated(commitment),
-				);
-
-				powRecords = powRecords.filter((commitment) =>
-					commitmentUpdated(commitment),
-				);
-			}
+			const BATCH_SIZE = 1000;
+			const captchaDB = new CaptchaDatabase(
+				this.config.mongoCaptchaUri,
+				undefined,
+				undefined,
+				this.logger,
+			);
 
-			if (commitments.length || powRecords.length) {
-				this.logger.info(
-					`Storing ${commitments.length} commitments externally`,
-				);
+			// Process image commitments with cursor
+			let processedCommitments = 0;
+			await this.processBatchesWithCursor(
+				async (skip: number) =>
+					await this.providerDB.getUnstoredDappUserCommitments(
+						BATCH_SIZE,
+						skip,
+					),
+				async (batch) => {
+					const lastUpdated = lastTask?.updated || 0;
+					const filteredBatch = lastTask?.updated
+						? batch.filter((commitment) => this.isCommitmentUpdated(commitment))
+						: batch;
 
-				this.logger.info(
-					`Storing ${powRecords.length} pow challenges externally`,
-				);
+					if (filteredBatch.length > 0) {
+						await captchaDB.saveCaptchas(filteredBatch, []);
+						await this.providerDB.markDappUserCommitmentsStored(
+							filteredBatch.map((commitment) => commitment.id),
+						);
+					}
+					processedCommitments += filteredBatch.length;
+				},
+			);
 
-				const captchaDB = new CaptchaDatabase(
-					this.config.mongoCaptchaUri,
-					undefined,
-					undefined,
-					this.logger,
-				);
+			// Process PoW records with cursor
+			let processedPowRecords = 0;
+			await this.processBatchesWithCursor(
+				async (skip: number) =>
+					await this.providerDB.getUnstoredDappUserPoWCommitments(
+						BATCH_SIZE,
+						skip,
+					),
+				async (batch) => {
+					const lastUpdated = lastTask?.updated || 0;
+					const filteredBatch = lastTask?.updated
+						? batch.filter((record) => this.isCommitmentUpdated(record))
+						: batch;
 
-				await captchaDB.saveCaptchas(commitments, powRecords);
+					if (filteredBatch.length > 0) {
+						await captchaDB.saveCaptchas([], filteredBatch);
+						await this.providerDB.markDappUserPoWCommitmentsStored(
+							filteredBatch.map((record) => record.challenge),
+						);
+					}
+					processedPowRecords += filteredBatch.length;
+				},
+			);
 
-				await this.providerDB.markDappUserCommitmentsStored(
-					commitments.map((commitment) => commitment.id),
-				);
-				await this.providerDB.markDappUserPoWCommitmentsStored(
-					powRecords.map((powRecords) => powRecords.challenge),
-				);
-			}
 			await this.providerDB.updateScheduledTaskStatus(
 				taskID,
 				ScheduledTaskStatus.Completed,
 				{
 					data: {
-						commitments: commitments.map((c) => c.id),
-						powRecords: powRecords.map((pr) => pr.challenge),
+						processedCommitments,
```

Comment:
Can you change this to a a count please?

Reply:
It is a count